pkg/controller/config/config_controller.go

```diff
@@ -320,8 +322,14 @@ func (r *ReconcileConfig) applyAddons(req reconcile.Request, cfg *op.Config) (re
 	log := requestLogger(req, "apply-addons")
 
 	//add TaskProviderType label to ClusterTasks (community, redhat, certified)
-	injectLabel := transform.InjectLabel(flag.LabelProviderType, flag.ProviderTypeRedHat, transform.Overwrite, "ClusterTask")
-	if err := transformManifest(cfg, &r.addons, injectLabel); err != nil {
+	triggerImages := transform.ToLowerCaseKeys(imagesFromEnv("IMAGE_TRIGGERS_"))
+	addonImages := transform.ToLowerCaseKeys(imagesFromEnv("IMAGE_ADDONS_"))
+	addnTfrms := []mf.Transformer{
+		transform.InjectLabel(flag.LabelProviderType, flag.ProviderTypeRedHat, transform.Overwrite, "ClusterTask"),
+		transform.DeploymentImages(triggerImages),
```

Comment:
can we pass prefixes like these `"IMAGE_TRIGGERS_"`  to transformer and 
call transform.ToLowerCaseKeys(imagesFromEnv("IMAGE_TRIGGERS_")) from the transformer factory function

Reply:
sounds true, initially it was like that only. Later we decoupled it to use the map because of two reasons
1) So the transformation could be more tested by making map dependency, that we could inject it during the test
2) Source could be anything may be ENV,  some CR values. It would be easy to reuse while porting it for the upstream use case